src/pages/_app.js

```diff
@@ -96,6 +97,16 @@ const MyApp = (props) => {
       jssStyles.parentElement.removeChild(jssStyles)
     }
   }, [])
+  const [theme, setTheme] = useState(createTheme(defaultTheme))
+
+  //  make updater function in theme context referentially stable
+  const updater = useCallback((causeId) => setTheme(themeMapper(causeId)), [])
+
+  //  optimizer
+  const memoizedValue = useMemo(
+    () => ({ theme, setTheme: updater }),
+    [theme, updater]
+  )
```

Comment:
Some feedback on code composition: can you move this logic into your hook to set the default theme value and setter function there? Then, I believe _app.js can just call `useTheme` just like any other page and receive the default value because there's no provider in the tree above.

Reply:
This will help make it easier to understand what `setTheme` is doing when a page calls it, because the hook maintains the state.

Reply:
Never mind, I don't think my feedback makes sense.